[CI] Remove `DOCTRINE_ORM_VERSION` from packages workflow

Rafikooo opened the pull request:
| Q               | A
|-----------------|-----
| Branch?         | 2.1
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

1. Removes `DOCTRINE_ORM_VERSION` support from packages CI workflow as we no longer use it.
2. Refactors `RoboFile` slightly.

GSadee commented on .github/workflows/ci_e2e-unstable.yaml:
```diff
@@ -31,6 +31,7 @@ jobs:
         env:
             APP_ENV: test_cached
             DATABASE_URL: "mysql://root:root@127.0.0.1/sylius?charset=utf8mb4&serverVersion=${{ matrix.mysql }}"
+            DOCTRINE_ORM_VERSION: "${{ matrix.orm }}"
```
If I am not mistaken, these variables are not defined anywhere. I would be in favour of removing setting here the ORM version and adjusting the RoboFile file, which is responsible for building packages tests

Rafikooo committed:
Remove DOCTRINE_ORM_VERSION support from packages CI

- Remove ORM version environment variable and related logic
- Fix GitHub Actions group closing (move endGroup inside loop)
- Add RuntimeException import and remove unnecessary qualifiers
- Simplify exit condition and remove redundant null check
